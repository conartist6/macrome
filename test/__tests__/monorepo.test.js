const { testProject } = require('./test-project');
const { sandboxPath, gitStatus, eventually } = require('./utils');

const { writeFile, readFile, unlink } = require('fs').promises;

describe('monorepo', () => {
  let testA, testB;

  describe('project a', () => {
    testA = testProject(sandboxPath('projects/monorepo/project-a'));
  });

  describe('project b', () => {
    testB = testProject(sandboxPath('projects/monorepo/project-b'));
  });

  describe('projects a and b', () => {
    testA.watchSetup();
    testB.watchSetup();

    beforeAll(() => {
      process.chdir(sandboxPath('projects/monorepo'));
    });

    it('watch', async () => {
      const aPath = 'project-a/lib/project-a.js';
      const bPath = 'project-b/lib/project-b.js';
      const aContent = await readFile(aPath, 'utf8');
      const bContent = await readFile(bPath, 'utf8');

      await Promise.all([unlink(aPath), unlink(bPath)]);

      await eventually(() =>
        expect(gitStatus()).toMatchInlineSnapshot(`
          Array [
            " D project-a/lib/generated-project-a.js",
            " D project-a/lib/project-a.js",
            " D project-b/lib/generated-project-b.js",
            " D project-b/lib/project-b.js",
          ]
        `),
      );

      await Promise.all([writeFile(aPath, aContent + '\n'), writeFile(bPath, bContent + '\n')]);

      await eventually(() => {
        expect(gitStatus()).toMatchInlineSnapshot(`
          Array [
            " M project-a/lib/generated-project-a.js",
            " M project-a/lib/project-a.js",
            " M project-b/lib/generated-project-b.js",
            " M project-b/lib/project-b.js",
          ]
        `);
      });

      await Promise.all([writeFile(aPath, aContent), writeFile(bPath, bContent)]);

      await eventually(() => {
        expect(gitStatus()).toMatchInlineSnapshot(`Array []`);
      });
    });
  });
});
