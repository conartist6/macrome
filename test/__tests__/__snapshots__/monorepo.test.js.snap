// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`monorepo project a cleans 1`] = `
Array [
  " D lib/generated-project-a.js",
]
`;

exports[`monorepo project b cleans 1`] = `
Array [
  " D lib/generated-project-b.js",
]
`;

exports[`monorepo projects a and b watch 1`] = `
Array [
  " D project-a/lib/generated-project-a.js",
  " D project-a/lib/project-a.js",
  " D project-b/lib/generated-project-b.js",
  " D project-b/lib/project-b.js",
]
`;

exports[`monorepo projects a and b watch 2`] = `
Array [
  " M project-a/lib/generated-project-a.js",
  " M project-a/lib/project-a.js",
  " M project-b/lib/generated-project-b.js",
  " M project-b/lib/project-b.js",
]
`;
